tile_classes.py

```diff
@@ -21,38 +17,73 @@ def __init__(self,dim,tile_type,with_colors): # creating the table
+                                        y += 1
+                                        self.mod_char(x, y, item["character"])
+                                    else:
+                                        y -= 1
+                                        self.mod_char(x, y, item["character"])
+                self.mod_col(1, self.tile_elements[1]["character"]) # creating vertical edges of mountains on the world
+                self.mod_col(dim, self.tile_elements[1]["character"])
+                self.mod_row(dim, self.tile_elements[1]["character"])
+                self.mod_row(1, self.tile_elements[1]["character"])
+
+            elif self.is_custom == True: # creating world if player decides to use a custom map
+                custom_map = open("custom_map.txt")
+                for line in custom_map: # converting text file to list information that is the world
+                    self.tile.append(list(line.strip("\n")))
+
+                if len(self.tile[0]) != len(self.tile): # making sure custom tile is square
+                    print("Invalid custom map dimensions: tile must be square!")
+                    quit()
+                else: # storing dim if the tile is square
+                    self.tile_dim = len(self.tile)
+
+                if self.with_colors == True: # updating characters from stored map if game is loaded with colors
+                    row_pos = self.tile_dim
+                    for row in self.tile:
+                        col_pos = 1
+                        for character in row:
+                            if character == ".":
+                                self.mod_char(col_pos,row_pos,self.tile_elements[0]["character"])
+                            elif character == "M":
+                                self.mod_char(col_pos, row_pos, self.tile_elements[1]["character"])
+                            elif character == "^":
+                                self.mod_char(col_pos, row_pos, self.tile_elements[2]["character"])
+                            elif character == " ":
+                                self.mod_char(col_pos,row_pos, " ")
                             else:
-                                if randint(0,1) == True:
-                                    y += 1
-                                    self.mod_char(x, y, item["character"])
-                                else:
-                                    y -= 1
-                                    self.mod_char(x, y, item["character"])
-            self.mod_col(1, self.tile_elements[1]["character"]) # creating vertical edges of mountains on the world
-            self.mod_col(dim, self.tile_elements[1]["character"])
-            self.mod_row(dim, self.tile_elements[1]["character"])
-            self.mod_row(1, self.tile_elements[1]["character"])
+                                self.mod_char(col_pos, row_pos, self.tile_elements[1]["character"]) # converting unrecognized characters to mountains
```

Comment:
Why are you "converting unrecognized characters to mountains"?

Reply:
Because they have to be converted into something or the program kills itself, and mountains seemed like the best option since they're non-traversable by players

Reply:
@amistele Make sure this behavior is documented... if in the future we encounter a bug when new/different characters are used